core/templates/dev/head/domain/utilities/LanguageUtilService.js

```diff
@@ -18,19 +18,35 @@
 
 oppia.factory('LanguageUtilService', [
   'AudioLanguageObjectFactory', 'AutogeneratedAudioLanguageObjectFactory',
-  'BrowserCheckerService', 'AUTOGENERATED_AUDIO_LANGUAGES',
-  'SUPPORTED_AUDIO_LANGUAGES',
+  'BrowserCheckerService', 'ALL_LANGUAGE_CODES',
+  'AUTOGENERATED_AUDIO_LANGUAGES', 'SUPPORTED_AUDIO_LANGUAGES',
   function(
       AudioLanguageObjectFactory, AutogeneratedAudioLanguageObjectFactory,
-      BrowserCheckerService, AUTOGENERATED_AUDIO_LANGUAGES,
-      SUPPORTED_AUDIO_LANGUAGES) {
+      BrowserCheckerService, ALL_LANGUAGE_CODES,
+      AUTOGENERATED_AUDIO_LANGUAGES, SUPPORTED_AUDIO_LANGUAGES) {
     var supportedAudioLanguageList = SUPPORTED_AUDIO_LANGUAGES;
     var autogeneratedAudioLanguageList = AUTOGENERATED_AUDIO_LANGUAGES;
 
     var supportedAudioLanguages = {};
     var autogeneratedAudioLanguagesByExplorationLanguageCode = {};
     var autogeneratedAudioLanguagesByAutogeneratedLanguageCode = {};
 
+    var getShortLanguageDescription = function(fullLanguageDescription) {
+      ind = fullLanguageDescription.indexOf(' (');
+      if (ind === -1) {
```

Comment:
I am a bit surprised at the branching. What are examples that would fall in both these cases?

If we expect a certain format, it would be better to have a test validating that format so that we can assume it here.

(EDIT: I do realize that you copied this from the backend but it might still be worth taking a moment to double-check, I think, so that we don't write unused code.)

Reply:
I think that if we define the constant ourselves we should maybe have a field `shortDescription` and retrieve the value directly instead of slicing strings.

Reply:
@seanlip This is being used for language constant present [here](https://github.com/oppia/oppia/blob/develop/assets/constants.js#L93). If will be used for [this](https://github.com/oppia/oppia/blob/develop/assets/constants.js#L95). For all other languages, else will be used. Should I add `shortDescription` as suggested by @vojtechjelinek ?

Reply:
Ah, I see, thanks for checking. I'm fine with either leaving it as-is or doing what @vojtechjelinek selected. If we go the latter route, it would be nice to have a Karma test as well that reads constants.ALL_LANGUAGE_CODES and verifies that `shortDescription` is a substring of `description`.

Reply:
@vojtechjelinek I am confused how to proceed here. Should I add `shortDescription` field for language constants present [here](https://github.com/oppia/oppia/blob/develop/assets/constants.js#L93) or leave them as-is?

Reply:
Do as you wish.

Reply:
As `constants.ALL_LANGUAGE_CODES` is being used in many places in code, I have decided to leave it as-is. @seanlip PTAL, thanks.